### DISCUSSION TITLE
customize job data

### DISCUSSION THREAD

**Discussion by dryodryo** [audioscavenger/save-image-extended-comfyui#21] [Ideas]
Hi,

Thanks for maintaining and continuing development on this super useful node!

I would love to be able to grab any widget value and plug it into the .JSON job data file, just as we can with filename and folder keys.

The major issue currently is that save_image_extended job data doesn't record the unet_name from the Load Diffusion Model node. So, the .JSON file does not record the most critical information about the image, the diffusion model name, e.g. Flux.

Thankfully, I can use the badge ID functionality to plug the model name into the filename, but the .JSON job data needs this, too.

Thanks again

**Comment by audioscavenger** [comment]
thanks and Yes, that's why I went to the badge UI number to identify the nodes you want to save info of. There can be multiple of the same node and therefore the last one added would be the winner, always. This node has no idea what information is relevant to you and bade ID is the only way to work with that I can think of.
Now that you mention the JSON job saving part, I must admit that I left it untouched when forking. I have no use for that and not sure what should be saved in it. It seems that you want it to be a complete repository for the prompt/workflow, am I reading this correctly?

**Comment by dryodryo** [comment]
Thanks, I don't think you need to include the entire prompt/workflow in the SaveImageExtended JSON file. We already have that ability using the ComfyUI native file format, and in the metadata of rendered image. But it would be very helpful to store specific keys in the SaveImageExtended sidecar JSON file. I am a graphics professional and very much a stickler for proper asset management. All of the data I want to keep easily accessible is getting too long to be practical for a filename. If the SaveImageExtended sidecar JSON file could store arbitrary keys, that would be fantastic. Ideally it would be a "sidecar file keys" field, totally separate from the filename keys field. I have a kind of complicated system of manually creating delimiter characters. I don't use the built-in delimiter because it's too ... limiting. If the sidecar file just grabbed all of the keys from the filename key field, it would be very messy.

Thanks again

**Reply by audioscavenger** [reply]
I understand you want a sidecar json or xml or smth, that stores more info then what you want to see in the filename, but also less then the full workflow (already stored in your image's EXIF data anyways).
Then, how do you envision and where would you write down the specific keys you want in that sidecar? The node is already quite tall with all its options, I refuse to add another text field. 

Maybe a different syntax for the keys? such as `2.key.json` ? I guess that would be possible... if no *.json is in the filename prefix, then default keys are saved in the json as it is currently. And if there is at least one *.json, then **those ones** only are saved in the json. 

give me your input and I'll do it once I get a chance!
